WalletWasabi/Tor/Socks5/Pool/Circuits/OneOffCircuit.cs

```diff
@@ -5,19 +5,31 @@ namespace WalletWasabi.Tor.Socks5.Pool.Circuits
 	/// <summary>
 	/// Random Tor circuit for an HTTP request which should not be linked with any other HTTP request sent previously.
 	/// </summary>
-	public class OneOffCircuit : ICircuit
+	public class OneOffCircuit : ICircuit, IDisposable
```

Comment:
Why did you add this?  I thought OneOffCircuit is working OK.

Reply:
Good question, it works OK now. Anyway, if we add `IDisposable` here, then we can simplify 

https://github.com/zkSNACKs/WalletWasabi/blob/71da493f123aac3264245e5b9086fffd52e079b1/WalletWasabi/Tor/Socks5/TorTcpConnectionFactory.cs#L87

to be simply

```cs
bool allowRecycling = !useSsl;
```

This said it's a low-level detail but it means that we control circuit lifetimes using `ICircuit.IsAlive` property and that gives us more options.

Also we can go one step further and we can actually remove `OneOffCircuit` class and just use `PersonCircuit` (i.e. creating `PersonCircuit`, sending one HTTP request and disposing the circuit is the same as what we have now as `OneOffCircuit`). So for this reason it seems good to do it to me. It can be in a different PR though.